API/Repositories/Interfaces/IPlayerRepository.cs

```diff
@@ -42,80 +44,52 @@ public interface IPlayerRepository : IRepository<Player>
-    /// <param name="username"></param>
-    /// <returns></returns>
+    /// <remarks>Username is case insensitive</remarks>
+    /// <returns>A player id, or null if not found</returns>
     Task<int?> GetIdAsync(string username);
 
     /// <summary>
-    /// Returns the osu id that belongs to this player id
-    /// </summary>
-    /// <param name="id"></param>
-    /// <returns></returns>
-    Task<long> GetOsuIdAsync(int id);
-
-    /// <summary>
-    ///  Returns players that haven't been updated in the last 14 days,
-    ///  or players that have never been updated.
+    /// Gets the id of a player for the given osu! id
     /// </summary>
-    /// <returns></returns>
-    Task<IEnumerable<Player>> GetOutdatedAsync();
-
-    /// <summary>
-    /// Get all players that are missing an earliest global rank in any mode (but have a current rank in that mode)
-    /// </summary>
-    /// <returns></returns>
-    Task<IEnumerable<Player>> GetPlayersMissingRankAsync();
-
-    /// <summary>
-    /// Returns a collection of <see cref="PlayerRatingDTO"/> in order by rating descending
-    /// for the given mode.
-    /// </summary>
-    /// <param name="n">The number of items to return</param>
-    /// <param name="mode">The mode to get the ratings from</param>
-    Task<IEnumerable<PlayerRatingDTO>> GetTopRatingsAsync(int n, OsuEnums.Mode mode);
+    /// <returns>A player id, or null if not found</returns>
+    Task<int?> GetIdAsync(long osuId);
 
     /// <summary>
-    /// Fetches the username for the given osu id, if available
+    /// Gets the osu! id for the given player id
     /// </summary>
-    /// <param name="osuId">The osu id</param>
-    /// <returns></returns>
-    Task<string?> GetUsernameAsync(long osuId);
+    /// <returns>An osu! id, or null if not found</returns>
+    Task<long> GetOsuIdAsync(int id);
 
     /// <summary>
-    /// Fetches the username for the given player id
+    /// Gets the username of a player for the given id
     /// </summary>
-    /// <param name="id">The user id</param>
+    /// <returns>A username, or null if not found</returns>
     Task<string?> GetUsernameAsync(int id);
 
     /// <summary>
-    /// Returns a collection of <see cref="PlayerIdMappingDTO"/> for all players
-    /// </summary>
-    Task<IEnumerable<PlayerIdMappingDTO>> GetIdMappingAsync();
-
-    /// <summary>
-    /// Returns a collection of <see cref="PlayerCountryMappingDTO"/> for all players
+    /// Gets the country of a player for the given id
     /// </summary>
-    Task<IEnumerable<PlayerCountryMappingDTO>> GetCountryMappingAsync();
+    /// <returns>A country code, or null if not found</returns>
+    Task<string?> GetCountryAsync(int id);
 
     /// <summary>
-    /// Returns the player id for the given user id
+    /// Gets all players that haven't been updated in the last 14 days
+    /// or have never been updated
     /// </summary>
-    /// <param name="userId"></param>
-    Task<int> GetIdAsync(int userId);
```

Comment:
This introduces a bug. This causes the service method to use the `GetAsync(long osuId)` repository method instead of the correct `int id` before. Why was this removed?
```cs
// PlayerService.cs L:23-24 
public async Task<PlayerDTO?> GetAsync(int id) =>
        _mapper.Map<PlayerDTO?>(await _playerRepository.GetAsync(id)); // !!calling long osuId overload!!
```

This causes the `playerInfo` field in `PlayerStatsService` to be null, breaking user profiles.